### PR TITLE
Add Markdown formatting page

### PR DESCRIPTION
**Pull request by Andre601**
Adds a page listing all supported Markdown options for Modrinth Projects and versions.

I would appreciate it if some people could check locally if everything formats properly. I didn't notice any real problems, but you never know...

### CODE REVIEW

File: docs/tutorials/markdown.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# Formatting
+
+Modrinth allows the usage of [GitHub Flavourited Markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) for text fields such as your Project's Description or Version description.
+
+To help you understand what is and isn't available to use is this page here listing all currently supported Markdown features.
```
Comment (wafflecoffee) [inline docs/tutorials/markdown.md:5]:
```suggestion
As the full GFM spec is too large to read comfortably and [GitHub's formatting guide](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) contains features not included in GFM, this page serves as a quick reference for what types of formatting can be used on Modrinth.
```

File: docs/tutorials/markdown.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+## Basic formatting
+
+| Markdown                    | Result                |
+| --------------------------- | --------------------- |
+| \*\*Bold\*\*                | **Bold**              |
+| \*Italic\*                  | *Italic*              |
+| \*\*\*Bold and Italic\*\*\* | ***Bold and italic*** |
+| \~\~Strikethrough\~\~       | ~~Strikethrough~~     |
+
+## Code Blocks
```
Comment (wafflecoffee) [inline docs/tutorials/markdown.md:16]: Headings should all be uncapitalised besides the first letter of the first word

File: docs/tutorials/markdown.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+\`\`\`  
+Code here  
+\`\`\`
+
+```
+Code here
+```
+
+:::
```
Comment (wafflecoffee) [inline docs/tutorials/markdown.md:36]: Either pick admonitions or tables, don't use both.

File: docs/tutorials/markdown.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+:::tip
+
+You can add a coding language (i.e. `java`) after the first three backticks to add code highlighting to the text inside the code block.
```
Comment (wafflecoffee) [inline docs/tutorials/markdown.md:40]:
```suggestion
You can add a language (e.g. `java`) after the first three backticks to add code highlighting to the code block.
```

**Review by wafflecoffee — CHANGES_REQUESTED** [review]
The file needs to be added to sidebars.js.

Overall I'm somewhat disappointed with how this has been formatted - maybe something better/more stylish can be done via the use of [MDX](https://docusaurus.io/docs/markdown-features/react). Something like [XenForo's built-in BB Code guide](https://empireminecraft.com/help/bb-codes) would look nice.

File: docs/tutorials/markdown.md (hunk 2 above)
Comment (Andre601) [reply docs/tutorials/markdown.md]: Changed

File: docs/tutorials/markdown.md (hunk 3 above)
Comment (Andre601) [reply docs/tutorials/markdown.md:59]: I changed all examples to be in Admonitions and also made the raw version to be in code blocks.

**Comment by Andre601** [comment]
Friendly poke to remind that this PR still awaits a fresh review. :)

**Comment by wafflecoffee** [comment]
Could definitely look better if stuff was side-by-side via some MDX stuff, but good enough for now.

**Pull request merged**